### DISCUSSION TITLE
Atomic swap issue

### DISCUSSION THREAD

**Discussion by vision792** [comit-network/xmr-btc-swap#944] [Q&A] [answered]
![1456](https://user-images.githubusercontent.com/99592522/153753751-d12a917b-a1b2-4c96-b7aa-6071347412b3.jpg)
Hi,
I need your help.
Could someone tell me how I cancel and get a refund?
I tried the following:
./refund btc-address-- [...]
./swap cancel -- [...]
./swap refund -f --swap-id [...]
./swap withdraw-btc -- [...]
Even trying to get my swap id is not successful.
My mistake at the beginning is not having allowed .exe to communicate on Private networks.
Sorry for being newb.

**Comment by binarybaron** [comment]
Can you run `swap.exe history`

**Comment by vision792** [comment]
No :(

swap.exe : The term 'swap.exe' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again.
At line:1 char:1
+ swap.exe history
+ ~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (swap.exe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException


Suggestion [3,General]: The command swap.exe was not found, but does exist in the current location. Windows PowerShell does not load commands from the current location by default. If you trust this command, instead type: ".\swap.exe". See "get-help about_Command_Precedence" for more details.

**Reply by binarybaron** [reply]
Can you try `.\swap.exe history`

**Reply by vision792** [reply]
that one works

**Reply by binarybaron** [reply]
What is the output? You will have to resume your swap otherwise you might loose your funds

**Comment by vision792** [comment]
+--------------------------------------+-----------------+
| SWAP ID                              | STATE           |
+========================================================+
| bbee99bb-e056-468d-890d-b46f48aca490 | btc is redeemed |
+--------------------------------------+-----------------+

**Comment by binarybaron** [comment, marked as answer]
Looks good. Please resume the swap with `.\swap.exe resume --swap-id bbee99bb-e056-468d-890d-b46f48aca490`

**Comment by vision792** [comment]
that's what I get:

Failed to generate monero wallet from keys: Failed to generate new wallet from keys: JSON-RPC request failed with code -1: Wallet already exists.: JSON-RPC request failed with code -1: Wallet already exists.
Falling back to trying to open the the wallet if it already exists ←[3mwallet_file_name←[0m←[2m=←[0mbbee99bb-e056-468d-890d-b46f48aca490
Connected to Alice at /ip4/51.15.44.142/tcp/9939/p2p/12D3KooWAPci3x4GuWDi9P86efnR1q9a7phKnPwnwQcRNMW2HDUy

**Comment by vision792** [comment]
Lost connection to Alice at /ip4/51.15.44.142/tcp/9939/p2p/12D3KooWAPci3x4GuWDi9P86efnR1q9a7phKnPwnwQcRNMW2HDUy, cause: Connection error: I/O error: connection is closed
Error: JSON-RPC request failed with code -4: Failed to get height

Caused by:
    JSON-RPC request failed with code -4: Failed to get height

**Comment by vision792** [comment]
That's what I get. Is there hope I could get back my funds?

**Reply by binarybaron** [reply]
Yes, definitely. You are not at risk of loosing your funds, there is just an issue with the creation of redeem wallet (not sure why). You should be able to manually redeem the Monero by running `.\swap.exe monero-recovery --swap-id bbee99bb-e056-468d-890d-b46f48aca490`. It will print all the information you need to use the Monero GUI to create a wallet to then redeem your funds (create a new wallet and use the primary address and private key that were printed to create the wallet)

**Keep the output of that command private! (Don't paste it into here)**
